<?xml version="1.0"?>
<!-- author: Pierre Joyet (3407684) - Marine CORNET (3531423) - ROS Project - M1 SAR 2020 -->

<launch>
  <!-- gazebo gui launch -->
  <arg name="gui" default="true"/>

  <!-- challenge 3 task 1 running -->
  <arg name="task1" default="true"/>

  <!-- if task 1 running True -->
  <group if="$(arg task1)">
    <!-- launch task 1 -->
    <include file="$(find challenge_project)/launch/challenge3/challenge3_task1.launch"/>
  </group>

  <!-- if task 1 running False -->
  <group unless="$(arg task1)">
    <!-- set running task 1 indicator to False -->
    <param name="task1_running" type="bool" value="false"/>
    <!-- robot initial position and rotation -->
    <include file="$(find challenge_project)/launch/gazebo.launch">
      <arg name="x" value="-5.2"/>
      <arg name="y" value="1.2"/>
      <arg name="z" value="0.02"/>
      <arg name="roll" value="0"/>
      <arg name="pitch" value="0"/>
      <arg name="yaw" value="3.14"/>
      <arg name="world" value="$(find challenge_project)/worlds/challenge3.world"/>
      <arg name="gui" value="$(arg gui)"/>
    </include>
    <!-- start distance publisher node -->
    <node pkg="challenge_project" name="lds_distance" type="lds_distance_publisher.py" required='true'/>

    <!-- camera image viewer -->
    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
      <remap from="image" to="/camera/image_raw" />
    </node>

  </group>

  <!-- start distance publisher node -->
  <param name="task2_part1_running" type="bool" value="true"/>
  <param name="task2_part2_running" type="bool" value="true"/>

  <!-- speed value -->
  <param name="linear_scale" type="double" value="0.5" />

  <!-- start movement corridor part 1 and 2 node -->
  <node pkg="challenge_project" name="movement_corridor" type="movement_corridor.py"/>
  <node pkg="challenge_project" name="movement_corridor_2" type="movement_corridor_part2.py"/>



</launch>
